// @sports-alliance/sports-lib/lib/meta-data/meta-data.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GarminHealthAPIEventMetaData = exports.COROSAPIEventMetaData = exports.SuuntoAppEventMetaData = void 0;
var event_meta_data_interface_1 = require("./event-meta-data.interface");
var SuuntoAppEventMetaData = /** @class */ (function () {
    function SuuntoAppEventMetaData(serviceWorkoutID, serviceUserName, date) {
        this.serviceWorkoutID = serviceWorkoutID;
        this.serviceUserName = serviceUserName;
        this.date = date;
        this.serviceName = event_meta_data_interface_1.ServiceNames.SuuntoApp;
    }
    SuuntoAppEventMetaData.prototype.toJSON = function () {
        return {
            serviceWorkoutID: this.serviceWorkoutID,
            serviceName: this.serviceName,
            serviceUserName: this.serviceUserName,
            date: this.date.getTime()
        };
    };
    return SuuntoAppEventMetaData;
}());
exports.SuuntoAppEventMetaData = SuuntoAppEventMetaData;
var COROSAPIEventMetaData = /** @class */ (function () {
    function COROSAPIEventMetaData(serviceWorkoutID, serviceOpenId, serviceFITFileURI, date) {
        this.serviceWorkoutID = serviceWorkoutID;
        this.serviceOpenId = serviceOpenId;
        this.serviceFITFileURI = serviceFITFileURI;
        this.date = date;
        this.serviceName = event_meta_data_interface_1.ServiceNames.COROSAPI;
    }
    COROSAPIEventMetaData.prototype.toJSON = function () {
        return {
            serviceWorkoutID: this.serviceWorkoutID,
            serviceName: this.serviceName,
            serviceOpenId: this.serviceOpenId,
            serviceFITFileURI: this.serviceFITFileURI,
            date: this.date.getTime()
        };
    };
    return COROSAPIEventMetaData;
}());
exports.COROSAPIEventMetaData = COROSAPIEventMetaData;
var GarminHealthAPIEventMetaData = /** @class */ (function () {
    function GarminHealthAPIEventMetaData(serviceUserID, serviceActivityFileID, serviceActivityFileType, serviceManual, serviceStartTimeInSeconds, date) {
        this.serviceUserID = serviceUserID;
        this.serviceActivityFileID = serviceActivityFileID;
        this.serviceActivityFileType = serviceActivityFileType;
        this.serviceManual = serviceManual;
        this.serviceStartTimeInSeconds = serviceStartTimeInSeconds;
        this.date = date;
        this.serviceName = event_meta_data_interface_1.ServiceNames.GarminHealthAPI;
    }
    GarminHealthAPIEventMetaData.prototype.toJSON = function () {
        return {
            serviceUserID: this.serviceUserID,
            serviceName: this.serviceName,
            serviceActivityFileID: this.serviceActivityFileID,
            serviceActivityFileType: this.serviceActivityFileType,
            serviceManual: this.serviceManual,
            serviceStartTimeInSeconds: this.serviceStartTimeInSeconds,
            date: this.date.getTime()
        };
    };
    return GarminHealthAPIEventMetaData;
}());
exports.GarminHealthAPIEventMetaData = GarminHealthAPIEventMetaData;
